service/api/service.go

```diff
@@ -454,19 +462,23 @@ func (s *serviceImpl) tryPrepareRPCbyDescribe(ctx context.Context, req iwfidl.Wo
 	var searchAttributes []iwfidl.SearchAttribute
 	var dataAttributes []iwfidl.KeyValue
 
-	requestedSAs := req.SearchAttributes
+	var requestedSAs []iwfidl.SearchAttributeKeyAndType
 	saPolicy := req.GetSearchAttributesLoadingPolicy()
-	if saPolicy.GetPersistenceLoadingType() != iwfidl.ALL_WITHOUT_LOCKING {
+	switch saPolicy.GetPersistenceLoadingType() {
+	case iwfidl.PARTIAL_WITH_EXCLUSIVE_LOCK, iwfidl.PARTIAL_WITHOUT_LOCKING:
 		requestedSAKeys := map[string]bool{}
 		for _, saKey := range saPolicy.PartialLoadingKeys {
 			requestedSAKeys[saKey] = true
 		}
-		requestedSAs = []iwfidl.SearchAttributeKeyAndType{}
 		for _, sa := range req.SearchAttributes {
 			if requestedSAKeys[sa.GetKey()] {
 				requestedSAs = append(requestedSAs, sa)
 			}
 		}
+	case iwfidl.NONE:
+		requestedSAs = []iwfidl.SearchAttributeKeyAndType{}
+	default:
```

Comment:
Nit: I think it's safer to explicitly handle the `ALL_WITHOUT_LOCKING` and return error for other cases that it doesn't understand